src/appleseed.shaders/src/npr/as_toon.osl

```diff
@@ -0,0 +1,316 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Esteban Tovagliari, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include "appleseed/color/as_color_blend_modes.h"
+
+//
+// Reference:
+//
+//
+
+shader as_toon
+[[
+    string help = "XToon material node.",
+    string icon = "asDisneyMaterial.png",
+
+    string as_maya_node_name = "asToon",
+    string as_maya_classification = "drawdb/shader/surface:rendernode/appleseed/surface:shader/surface:swatch/AppleseedRenderSwatch",
+    int as_maya_type_id = 0x7fffe, // Temporary Maya TypeId.
```

Comment:
Make type IDs official?

Reply:
TypeID = 0x00127a01, https://github.com/appleseedhq/appleseed-maya/pull/165/commits/e8e7ba5c7d46108e6e4d9665cd8a844ce6fd80a3#diff-698ace81a81189755ac0a39843c0d475R106

Reply:
Thanks! The asXToon shader also needs one IIRC.

Reply:
Got it, XToon type ID 0x00127a02
https://github.com/appleseedhq/appleseed-maya/pull/165/commits/98f49e5224b68df2b2dd55c02723394b4a891320

Reply:
I consider the shader experimental still and that is why they don't have fixed typeids and they are in their own folder. They don't show by default in Maya or other apps.

The XToon shader in particular is very flexible but I found it hard to use in its current form.